src/wallet/db.cpp

```diff
@@ -826,6 +826,8 @@ void BerkeleyDatabase::Flush(bool shutdown)
             LOCK(cs_db);
             g_dbenvs.erase(env->Directory().string());
             env = nullptr;
+        } else {
```

Comment:
That would cause an access violation on Windows since it's a wild pointer in the second call. Also, since the env is deleted when `g_dbenvs.erase` called, the map doesn't exist anymore.